docs/link-expansion.md

```diff
@@ -25,22 +25,13 @@
 
 ## Introduction
 
-Link expansion is a concept in the Publishing API which describes the process
-of converting the stored links of an edition into a JSON representation
-containing the details of these links. It is used in the Publishing API
-during the process of sending an edition downstream to the
-[Content Store][content-store].
+Publishing API stores links between content items so that frontend applications can render links to related content on GOV.UK, such as the organisation responsible for publishing the page. These links contain only a reference to the related content item - they don't contain any information about the related content item until they are "expanded".
 
-The process involves determining which editions should and can be linked
-to, the versions of them that are linked, and the fields that will be included
-in the representation. It replaces a process in the Content Store which was
-used to determine and expand links at the point of request.
+Link expansion is the process of converting the stored links of an edition into a JSON representation containing more information about the linked content items. The expansion process occurs when sending an edition downstream to the
+[Content Store][content-store].
 
 A closely related process to this is
-[dependency-resolution](dependency-resolution.md). This is something of a
-reversal in link expansion and has the role of determining which editions
-will need re-presenting to the content store as a result of an update to a
-document.
+[dependency resolution](dependency-resolution.md). This is the opposite of link expansion. When a document is updated, Publishing API works out which documents link to the updated document, and presents the linked documents to Content Store.
```

Comment:
I wanted to add something here about the limitations of dependency resolution in cases where the cardinality of the links has changed or the ordering of the links, but I couldn't actually rationalise what the limitation is.

Theoretically when an edition has a new link added to it, dependency resolution should run to ensure that any content items which link to the edition are presented to content store with their latest expanded links. Is that not how things work in reality?

Obviously when it's a new document we can't have linked to it yet so dependency resolution can't work, but that can be managed through Publishing apps (though at this point if Publishing apps are doing callbacks to update related items on create, they might as well do them on update and delete as well, so dependency resolution isn't abstracting away a lot of complexity for them).